tests/tensor/test_basic.py

```diff
@@ -2386,194 +2386,119 @@ def test_is_flat():
     assert not ptb.is_flat(X.reshape((iscalar(),) * 3))
 
 
-def test_tile():
-    """
-    TODO FIXME: Split this apart and parameterize.  Also, find out why it's
-    unreasonably slow.
-    """
+class TestTile:
+    @pytest.mark.parametrize(
+        "A_shape, reps_test",
+        [
+            ((), (2,)),
+            ((5,), (2,)),
+            ((2, 4), (2, 3)),
+            ((2, 4), (2, 3, 4)),
+            ((2, 4, 3), (2, 3)),
+            ((2, 4, 3), (2, 3, 4)),
+            ((2, 4, 3, 5), (2, 3, 4, 6)),
+        ],
+    )
+    def test_tile_separate_reps_entries(self, A_shape, reps_test):
+        rng = np.random.default_rng(2400)
 
-    def run_tile(x, x_, reps, use_symbolic_reps):
-        if use_symbolic_reps:
-            rep_symbols = [iscalar() for _ in range(len(reps))]
-            f = function([x, *rep_symbols], tile(x, rep_symbols))
-            return f(*([x_, *reps]))
-        else:
-            f = function([x], tile(x, reps))
-            return f(x_)
+        A = tensor("A", shape=(None,) * len(A_shape))
+        reps = [iscalar(f"r{i}") for i in range(len(reps_test))]
+        tile_out = tile(A, reps)
 
-    rng = np.random.default_rng(utt.fetch_seed())
+        tile_fn = function([A, *reps], tile_out)
 
-    for use_symbolic_reps in [False, True]:
-        # Test the one-dimensional case.
-        x = vector()
-        x_ = rng.standard_normal(5).astype(config.floatX)
-        assert np.all(run_tile(x, x_, (2,), use_symbolic_reps) == np.tile(x_, (2,)))
+        A_test = rng.standard_normal(A_shape).astype(config.floatX)
+        np.testing.assert_array_equal(
+            tile_fn(A_test, *reps_test),
+            np.tile(A_test, reps_test),
+            strict=True,
+        )
 
-        # Test the two-dimensional case.
-        x = matrix()
-        x_ = rng.standard_normal((2, 4)).astype(config.floatX)
-        assert np.all(run_tile(x, x_, (2, 3), use_symbolic_reps) == np.tile(x_, (2, 3)))
-
-        # Test the three-dimensional case.
-        x = tensor3()
-        x_ = rng.standard_normal((2, 4, 3)).astype(config.floatX)
-        assert np.all(
-            run_tile(x, x_, (2, 3, 4), use_symbolic_reps) == np.tile(x_, (2, 3, 4))
+    @pytest.mark.parametrize("reps", (2, np.array([2, 3, 4])))
+    def test_combined_reps_entries(self, reps):
+        rng = np.random.default_rng(2422)
+        A_test = rng.standard_normal((2, 4, 3)).astype(config.floatX)
+        expected_eval = np.tile(A_test, reps)
+
+        A = tensor3("A")
+        np.testing.assert_allclose(
+            tile(A, reps).eval({A: A_test}),
+            expected_eval,
         )
```

Comment:
There is still an `assert_allclose` here, if you want to use `assert_array_equal(strict=True)`, otherwise I think it's good to go

Reply:
Thanks, missed it